File: IT/Containers/Esercizi/Java/Esercizio14.java
/* file Esercizio14.java */

import java.util.List;
import java.util.ArrayList;
import java.util.Set;
import java.util.HashSet;
import java.util.Random;

public class Esercizio14 {
	public static void main(String [] args) {
		Random rand = new Random();
		List<Integer> nums = new ArrayList<Integer>();

		for (int i = 0; i < 10; i++) {
			int r = rand.nextInt(10) + 1;
			nums.add(r);
		}

		System.out.println("Lista originale: ");
		System.out.println(nums);

		Set<Integer> uniques = new HashSet<Integer>(nums);

		System.out.println("Lista senza elementi duplicati: ");
		System.out.println(uniques);
	}
}

File: IT/1/Containers/Esercizi/C++/esercizio14.cpp
/*
file esercizio14.cpp
compilare con: g++ -std=c++11 esercizio14.cpp
*/

#include <iostream>
#include <iterator>
#include <random>
#include <unordered_set>
#include <list>

using std::cout;
using std::endl;
using std::ostream_iterator;
using std::random_device;
using std::default_random_engine;
using std::uniform_int_distribution;
using std::list;
using std::unordered_set;

int main()
{
	list<int> nums;

	default_random_engine gen(random_device{}());
	uniform_int_distribution<int> dist(1, 10);

	for (int i = 0; i < 10; i++) {
		int r = dist(gen);
		nums.push_back(r);
	}

	cout << "Vettore originale: " << endl << "[";
	copy(nums.begin(), nums.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	unordered_set<int> uniques(nums.begin(), nums.end());

	cout << "Vettore senza elementi duplicati: " << endl << "[";
	copy(uniques.begin(), uniques.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;
}

